handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -239,34 +239,36 @@ protected void deallocate() {
 
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
-        try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
-        } finally {
-            readerLock.unlock();
-        }
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            try {
+                ssl = SSL.newSSL(context.ctx, !context.isClient());
+            } finally {
+                readerLock.unlock();
```

Comment:
hmm... I think you misunderstood @Scottmitch  (or I did). I think have the lock be obtained while still in the synchronized block is what @Scottmitch not liked. @Scottmitch correct me if I am wrong tho